superbit_lensing/galsim/mock_superBIT_data.py

```diff
@@ -393,16 +401,37 @@ def make_a_star(ud, wcs, affine, optics, sbparams, logprint):
     # This is still an x/y corrdinate
     uv_pos = affine.toWorld(image_pos)
 
-    # Draw star flux at random; based on distribution of star fluxes in real images
-    flux_dist = galsim.DistDeviate(ud, function = lambda x:x**-1.5, x_min = 6, x_max = 47586)
-    star_flux=flux_dist()
-    if sbparams.bandpass=='crates_adu_b':
-        star_flux*=0.8271672
+    # Draw star flux at random; based on either a semi-analytic distribution or GAIA stars
+    # Default to blue stars, which are plenty bright
+
+    index = obj_index - 1
+
+    if star_cat is not None:
+        if sbparams.bandpass=='crates_adu_shape':
+            star_flux = star_cat['bit_flux_shape'][index]
+
+        elif sbparams.bandpass=='crates_adu_b':
+            star_flux = star_cat['bit_flux_b'][index]
+
+        else:
+            star_flux = star_cat['bit_flux_b'][index]
+
+        star_flux*=sbparams.exp_time
+
+
+    else:
+        pud = np.random.default_rng()
```

Comment:
There is a `pud` being passed as an arg now - do you really want to rewrite it here?